Sources/System/FilePath/FilePathTemp.swift

```diff
@@ -0,0 +1,281 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+// MARK: - API
+
+public func withTemporaryPath<R>(basename: FilePath.Component,
+                                 _ body: (FilePath) throws -> R) throws -> R {
```

Comment:
Nit: please follow the [stdlib's line breaking conventions](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#line-breaking). 

```suggestion
public func withTemporaryPath<R>(
  basename: FilePath.Component,
  _ body: (FilePath) throws -> R
) throws -> R {
```

I cannot help questioning if this really needs to be a top-level function, rather than some static method on `FilePath`.

But more importantly: does this _really_ belong in swift-system?